src/backend/distributed/commands/table.c

```diff
@@ -1239,6 +1240,27 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				}
 			}
 
+			/*
+			 * We check for ADD COLUMN .. GENERATED .. AS IDENTITY expr
+			 * since it uses a sequence as an internal dependency
+			 * we should deparse the statement
+			 */
+			constraint = NULL;
+			foreach_ptr(constraint, columnConstraints)
+			{
+				if (constraint->contype == CONSTR_IDENTITY)
+				{
+					deparseAT = true;
+					useInitialDDLCommandString = false;
+
+					/* the new column definition will have no constraint */
+					ColumnDef *newColDef = copyObject(columnDefinition);
+					newColDef->constraints = NULL;
```

Comment:
We don't do this in the create_distributed_table path. Why do we do it here?

Reply:
This can also cause downstream issues, because it allows NULL values to be written, and alter_distributed_table fails when there are NULL values in a column with GENERATED AS IDENTITY because it brings back the constraint.